### PR TITLE
fix: retrieve default value for non-string parameters

### PR DESCRIPTION
**Pull request by Thomzoy**
Should solve #59 and [this issue](https://github.com/mkdocstrings/python/issues/8) from `mkdocstrings-python`

### CODE REVIEW

File: src/griffe/agents/nodes.py [hunk 1]
```diff
@@ -535,6 +535,7 @@ def parse__all__(node: NodeAssign) -> set[str]:  # noqa: WPS116,WPS120
         """
         return {elt.s for elt in node.value.elts}  # type: ignore[attr-defined]
 
+
```
Comment (pawamoy) [inline src/griffe/agents/nodes.py:538]: Useless change (and Black would not be happy), please remove :smile: 

```suggestion
```

File: src/griffe/agents/nodes.py [hunk 2]
```diff
@@ -1186,12 +1187,9 @@ def get_parameter_default(node: AST, filepath: Path, lines_collection: LinesColl
...
         return None
-    if isinstance(node, NodeConstant):
-        return repr(node.value)
-    if isinstance(node, NodeStr):
-        return repr(node.s)
-    if isinstance(node, NodeName):
-        return node.id
+    value_to_str = _node_value_map.get(type(node), None)
+    if value_to_str is not None:
+        return value_to_str(node)
```
Comment (pawamoy) [inline src/griffe/agents/nodes.py:1192]: Something simpler:

```suggestion
    with suppress(KeyError):
        return get_value(node)
```

And `from contextlib import suppress` must be added to the module imports :slightly_smiling_face:

**Review by pawamoy — CHANGES_REQUESTED** [review]
Thanks a lot!

Could you also add tests for this? You can get inspiration from `test_building_annotations_from_nodes` in `test_nodes.py`. Let me know if you need more hints.

**Comment by pawamoy** [comment]
We should test simple defaults like strings, integers, floats and booleans, but also more complex expressions like `DEFAULT_DICT["key1"]`, as you showed in #59 :slightly_smiling_face:

File: src/griffe/agents/nodes.py (hunk 1 above)
Comment (Thomzoy) [reply src/griffe/agents/nodes.py:538]: Actually black added this line !
I have version 21.9b

File: src/griffe/agents/nodes.py (hunk 2 above)
Comment (Thomzoy) [reply src/griffe/agents/nodes.py]: Good one !

File: src/griffe/agents/nodes.py (hunk 1 above)
Comment (pawamoy) [reply src/griffe/agents/nodes.py:539]: I see, my bad!

**Review by pawamoy — APPROVED** [review]
LGTM (CI good enough, will fix in other commits)!

Thanks again!

**Pull request merged**